cmd/habitat/node/node.go

```diff
@@ -0,0 +1,105 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+
+	"github.com/eagraf/habitat/cmd/habitat/api"
+	dataproxy "github.com/eagraf/habitat/cmd/habitat/data_proxy"
+	"github.com/eagraf/habitat/cmd/habitat/procs"
+	"github.com/eagraf/habitat/cmd/habitat/proxy"
+	"github.com/eagraf/habitat/cmd/sources"
+	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/eagraf/habitat/pkg/p2p"
+	"github.com/eagraf/habitat/structs/ctl"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/viper"
+)
+
+const (
+	ReverseProxyHost = "0.0.0.0"
+	ReverseProxyPort = "2041"
+
+	P2PPort = "6000"
```

Comment:
doesn't have to be done in this PR, i can also make this as a fix, but i realized we can use `net.Listen("127.0.0.1:0")` to have the kernel pick any port for us. then we can just see what port is being used.

Reply:
I think its ok to hardcode for now, I had another comment about port stuff so maybe we should talk about this next time